src/lib/exportDxf.ts

```diff
@@ -0,0 +1,227 @@
+import { isArray } from '@src/lib/utils'
+import { getOperationVariableName } from '@src/lib/operations'
+import type { EngineCommandManager } from '@src/lang/std/engineConnection'
+import type { KclManager } from '@src/lang/KclSingleton'
+import {
+  findOperationPlaneArtifact,
+  type StdLibCallOp,
+} from '@src/lang/queryAst'
+
+// Exports a sketch operation to DXF format
+export async function exportSketchToDxf(
+  operation: StdLibCallOp,
+  dependencies: {
+    engineCommandManager: EngineCommandManager
+    kclManager: KclManager
+    toast: {
+      error: (message: string, options?: any) => void
+      loading: (message: string) => string
+      success: (message: string, options?: any) => void
+      dismiss: (toastId: string) => void
+    }
+    uuidv4: () => string
+    base64Decode: (input: string) => ArrayBuffer | Error
+    browserSaveFile: (
+      blob: Blob,
+      filename: string,
+      toastId: string
+    ) => Promise<void>
+  }
+): Promise<{ success: boolean; error?: string }> {
```

Comment:
@max-mrgrsk Should we maybe the | Error pattern we use elsewhere here? 

```suggestion
): Promise<boolean | Error> {
```